lnwallet/channel.go

```diff
@@ -3258,17 +3372,51 @@ func (lc *LightningChannel) ProcessChanSyncMsg(
 		openedCircuits = commitDiff.OpenedCircuitKeys
 		closedCircuits = commitDiff.ClosedCircuitKeys
 
-	} else if remoteChainTip.height+1 != msg.NextLocalCommitHeight {
+	// There should be no other possible states as long as the commit chain
+	// can have at most two elements. If there is something is wrong.
+	default:
+		walletLog.Errorf("ChannelPoint(%v), sync failed: remote's "+
+			"next commit height is %v, while we believe it is %v!",
+			msg.NextLocalCommitHeight, remoteTipHeight)
+
 		if err := lc.channelState.MarkBorked(); err != nil {
 			return nil, nil, nil, err
 		}
-
-		// If we don't owe them a commitment, yet the tip of their
-		// chain isn't one more than the next local commit height they
-		// report, we'll fail the channel.
 		return nil, nil, nil, ErrCannotSyncCommitChains
 	}
 
+	// If we didn't have recovery options, then the final check cannot be
+	// performed, and we'll return early.
+	if !hasRecoveryOptions {
+		return updates, openedCircuits, closedCircuits, nil
+	}
+
+	// At this point we have determined that either the commit heights are
+	// in sync, or that we are in a state we can recover from. As a final
+	// check, we ensure that the commitment point sent to us by the remote
+	// is valid.
+	var commitPoint *btcec.PublicKey
+	switch {
+	case msg.NextLocalCommitHeight == remoteTailHeight+2:
```

Comment:
This line doesn't seem reachable, since first check in switch above is 
```
case msg.NextLocalCommitHeight > remoteTailHeight+1:
    return ErrCannotSyncCommitChains
```

Reply:
That's `msg.NextLocalCommitHeight > remoteTipHeight+1`. The case where we send a new commitment to the remote (new remote _tip_) that they did not receive, results in `msg.NextLocalCommitHeight == remoteTipHeight`. The case where they receive and ACK this commitment results in `msg.NextLocalCommitHeight == remoteTipHeight+1`. Higher commit heights will be invalid.

The case `msg.NextLocalCommitHeight == remoteTailHeight+2` can happen when the new remote tip is ACKed, but we didn't receive the ACK. In that case we have a pending remote commitment, meaning `remoteTipHeight == remoteTailHeight+1`.

Reply:
ahh i see, i misread tip as tail. thanks for the explanation :)